controllers/restaurantController.js

```diff
@@ -0,0 +1,165 @@
+import { Restaurant } from "../models/Restaurant.js";
+import mongoose from "mongoose";
+
+const getRestaurants = async function (req, res) {
+  try {
+    const queryObj = { ...req.query };
+    let query = Restaurant.find(queryObj);
+    const restaurants = await query;
+
+    if (restaurants.length === 0) {
+      return res.status(404).json({
```

Comment:
Please create something like enum for all of those http status codes and share this code with others